Automattic/wp-calypso
# [PBD interim fix] Convert <EligibilityWarnings> to TypeScript

#38141 is going to require a number of changes to the `<EligibilityWarnings>` component. Converting to TS since I need to understand the data structures involved and might as well document them while I'm here.

#### Changes proposed in this Pull Request

* Convert `<EligibilityWarnings>` components to ts
* Add `isKnownHoldType` to handle case where API returned an unknown hold (previously this would break the render)
* Add some jsdoc annotations to selectors so the types flow through to the component
* Add `supportUrl: null` to a few places since TS wasn't happy checking for the existence of the property here https://github.com/Automattic/wp-calypso/blob/aaa6703f1d3e2663ecb69bf795b9f9dabdf9aeac/client/blocks/eligibility-warnings/hold-list.tsx#L141

Question: have we got a better way of dealing with `translate` not returning a `string` when we know it will in this case?
https://github.com/Automattic/wp-calypso/blob/aaa6703f1d3e2663ecb69bf795b9f9dabdf9aeac/client/blocks/eligibility-warnings/index.tsx#L73
Maybe `<Banner>`'s `title` prop should just accept a `ReactChild`.

#### Testing Instructions

To see the `<EligibiliyWarnings>` component, pick a site that doesn't support AT and try to upload a plugin or theme, or change the hosting options (biz plan only)
